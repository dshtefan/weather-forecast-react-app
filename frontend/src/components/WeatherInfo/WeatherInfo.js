import React from 'react'
import './WeatherInfo.scss'

const WeatherInfo = ({ pressure, wind, humidity, lat, lon}) => {
  return (
    <div className="info-block">
      <div className="item-info">Pressure: {pressure} hPA</div>
      <div className="item-info">Wind: {wind} m/s</div>
      <div className="item-info">Humidity: {humidity}%</div>
      <div className="item-info">Coord: [{lat}, {lon}]</div>
    </div>
  );
};

export default WeatherInfo;