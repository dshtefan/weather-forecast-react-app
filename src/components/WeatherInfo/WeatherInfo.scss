.info-block {
  height: 170px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.item-info {
  font-size: 24px;
  font-weight: 300;
  height: 40px;
}