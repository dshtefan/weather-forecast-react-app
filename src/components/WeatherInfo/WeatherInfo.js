import React from 'react'
import './WeatherInfo.scss'

const WeatherInfo = ({ pressure, wind, humidity, lat, lon}) => {
  return (
    <div id="item-info">
      <div id="item-info-left">
        <div className="weather-info-line">Pressure:</div>
        <div className="weather-info-line">Wind:</div>
        <div className="weather-info-line">Humidity:</div>
        <div className="weather-info-line">Coord.:</div>
      </div>
      <div id="item-info-right">
        <div className="weather-info-line">{pressure} hPa</div>
        <div className="weather-info-line">{wind} m/s</div>
        <div className="weather-info-line">{humidity}%</div>
        <div className="weather-info-line">[{lat}, {lon}]</div>
      </div>
    </div>
  );
};

export default WeatherInfo;