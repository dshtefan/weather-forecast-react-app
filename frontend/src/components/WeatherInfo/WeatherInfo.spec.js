import React from 'react';
import WeatherInfo from "./WeatherInfo";

describe('WeatherInfo', () => {
  const props =
    {
      pressure: '997',
      humidity: '86',
      wind: '5',
      lat: '60',
      lon: '30'
    };

  describe('с получеными данными', () => {
    const weatherInfo = shallow(<WeatherInfo {...props} />);
    it ('отображается корректно', () => {
      expect(shallowToJson(weatherInfo)).toMatchSnapshot();
    });
  });

  describe('с неполучеными данными', () => {
    const weatherInfo = shallow(<WeatherInfo />);
    it ('отображается корректно', () => {
      expect(shallowToJson(weatherInfo)).toMatchSnapshot();
    });
  });
});