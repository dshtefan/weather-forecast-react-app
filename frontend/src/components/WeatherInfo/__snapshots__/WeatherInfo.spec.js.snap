// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo с неполучеными данными отображается корректно 1`] = `
<div
  className="info-block"
>
  <div
    className="item-info"
  >
    Pressure: 
     hPA
  </div>
  <div
    className="item-info"
  >
    Wind: 
     m/s
  </div>
  <div
    className="item-info"
  >
    Humidity: 
    %
  </div>
  <div
    className="item-info"
  >
    Coord: [
    , 
    ]
  </div>
</div>
`;

exports[`WeatherInfo с получеными данными отображается корректно 1`] = `
<div
  className="info-block"
>
  <div
    className="item-info"
  >
    Pressure: 
    997
     hPA
  </div>
  <div
    className="item-info"
  >
    Wind: 
    5
     m/s
  </div>
  <div
    className="item-info"
  >
    Humidity: 
    86
    %
  </div>
  <div
    className="item-info"
  >
    Coord: [
    60
    , 
    30
    ]
  </div>
</div>
`;
